### ISSUE TITLE
[idea] builtin clever-f feature

### ISSUE THREAD

**Issue by t9md** [t9md/atom-vim-mode-plus#851]
# Summary

- Improve `f`, `F`, `t`, `T`.
- 2nd `f` repeat previous `f{char}`. `F` for opposite direction.
  - Which means we no longer use `;` and `,`(we can use these key for different purpose:tada:)
- Search across lines.(pure-Vim-f search within line)
- Highlight current f-ing-char
  - User can determine consequence of `f`, `F` beforehand(reduce failure/retry rate).
- Personally I want call this feature "Lazy-f".

# Refs

- [rhysd/clever-f](https://github.com/rhysd/clever-f.vim)
- [deris/vim-shot-f](https://github.com/deris/vim-shot-f)
- [unblevable/quick-scope](https://github.com/unblevable/quick-scope)

**Comment by bassjacob** [comment]
does it compose with c, d etc. ? f to find a char, then c-f to delete until the next char? or do I use ; and , for that use case?

**Comment by t9md** [comment]
You cannot use cf or df to repeat f-char.
Once you type char other than f or F, lazy-repeat-find mode is reset.

**Comment by bassjacob** [comment]
that's fair, I thought that might have been the way you broke out of it. still sounds like a cool feature :smile:

**Comment by bogdan0083** [comment]
@t9md Hi! I've just checked your last [https://github.com/t9md/atom-vim-mode-plus/pull/852](merged pull request) and didn't find tuned-f for multiple lines. (right now it works only for the line we're on). Are you planning to add this as an option in the future? Thanks!

**Comment by t9md** [comment]
I originally planned, but changed-mind after I read README of  https://github.com/goldfeld/vim-seek.

Why not use `/`(search) if you want go **across** lines?
I can not find benefit to use `f` over `/` when you want search across lines.

**Comment by bogdan0083** [comment]
well, 2-char-find is not very useful for using on the same line (just my opinion), but quite useful to go to short distance on other lines (like 3-4 lines below or above, whatever). 

Search is fine to use for the whole file, but personally i don't use it for lines nearby (need to press enter). I would like to hear from other guys what's their approach is :)

**Comment by t9md** [comment]
OK I havent evaluated in recent experiment.
First I'll evaluate by implementing it.
Then if I can find some benefit for that feature, i might include it to vmp.
By enabling findAcrossLine like option

**Comment by bogdan0083** [comment]
Alright, thanks!

**Comment by t9md** [comment]
Released v1.2.0.
Feedback welcome: Why `findAcrossLines` is necessary(or useless) for you.

**Comment by thancock20** [comment]
So, a few days ago I started evaluating this new feature. I have `findCharsMax` set to 100, and `findConfirmByTimeout` at 500. `useSmartcaseForFind`, `highlightFindChar`, `reuseFindForRepeatFind`, and `findAcrossLines` are all `true`.

To be perfectly honest, I was skeptical at first that I would find much use for `findAcrossLines`, but I started using it so I could give my opinion.

But, I've actually found myself using it instead of [jumpy](https://atom.io/packages/jumpy) for quickly moving the cursor over the text. It seems to be faster than jumpy, because there is no need to switch context and type in the label letters to "jump".

I also think I'm able to use it faster than searching with `/` or `?`, because it happens in place. So, I don't have to take my eyes off the text to look at the mini editor.

All in all, I'm pleasantly surprised to find the feature useful. If you keep it, I'll quite likely uninstall jumpy.
